src/spclient.rs

```diff
@@ -359,13 +457,17 @@ impl SpClient {
         Ok(())
     }
 
-    pub fn set_fees(psbt: &mut Psbt, fee_rate: u32, payer: String) -> Result<()> {
-        let payer_vouts: Vec<u32> = match SilentPaymentAddress::try_from(payer.clone()) {
+    pub fn set_fees(psbt: &mut Psbt, fee_rate: Amount, payer: String) -> Result<()> {
+        // just take the first output that belong to payer
```

Comment:
I think we should also allow for the 'Use max' flow. In that case, the user spends all their selected outputs, and there is no change address, meaning none of the outputs belong to the payer. I suppose in that case, we can just call this function and select any of the random outputs as the 'payer' output?